pkg/morph/deploy/notary.go

```diff
@@ -871,51 +872,56 @@ func newCommitteeNotaryActor(b Blockchain, localAcc *wallet.Account, committee k
-// waiting for success. Caller should by default wait for the problem to be
-// fixed, and if not, retry.
-//
-// notaryDepositDeficiencyHandler must not be called from multiple routines in
-// parallel.
-type notaryDepositDeficiencyHandler = func(lackOfGAS bool)
-
 // Amount of GAS for the single local account's GAS->Notary transfer. Relatively
 // small value for fallback transactions' fees.
 var singleNotaryDepositAmount = big.NewInt(1_0000_0000) // 1 GAS
 
-// constructs notaryDepositDeficiencyHandler working with the specified
-// Blockchain and GAS/Notary balance of the given account.
-func initNotaryDepositDeficiencyHandler(ctx context.Context, l *zap.Logger, b Blockchain, localAcc *wallet.Account) (notaryDepositDeficiencyHandler, error) {
-	localActor, err := actor.NewSimple(b, localAcc)
-	if err != nil {
-		return nil, fmt.Errorf("init transaction sender from local account: %w", err)
-	}
+func autoReplenishNotaryBalance(ctx context.Context, l *zap.Logger, b Blockchain, localAcc *wallet.Account, chTrigger <-chan struct{}) {
+	l.Info("tracking Notary balance for auto-replenishment...")
 
-	notaryContract := notary.New(localActor)
-	gasContract := gas.New(localActor)
+	var err error
+	var localActor *actor.Actor
+	var notaryContract *notary.Contract
+	var gasContract *nep17.Token
+	var txMonitor *transactionGroupMonitor
 	localAccID := localAcc.ScriptHash()
 
-	// multi-tick context
-	transferTxMonitor := newTransactionGroupMonitor(localActor)
-	expirationTxMonitor := newTransactionGroupMonitor(localActor)
+	for {
+		select {
+		case <-ctx.Done():
+			l.Info("Notary balance tracker stopped by context", zap.Error(ctx.Err()))
+			return
+		case _, ok := <-chTrigger:
+			if !ok {
+				l.Info("Notary balance tracker stopped by closed block channel", zap.Error(ctx.Err()))
+				return
+			}
+		}
+
+		if localActor == nil {
+			localActor, err = actor.NewSimple(b, localAcc)
+			if err != nil {
+				l.Error("failed to init transaction sender from local account, will try again later", zap.Error(err))
+				continue
+			}
+
+			notaryContract = notary.New(localActor)
+			gasContract = gas.New(localActor)
+			txMonitor = newTransactionGroupMonitor(localActor)
+		}
 
-	return func(lackOfGAS bool) {
 		notaryBalance, err := notaryContract.BalanceOf(localAccID)
 		if err != nil {
 			l.Error("failed to read Notary balance of the local account, will try again later", zap.Error(err))
-			return
+			continue
 		}
 
-		gasBalance, err := gasContract.BalanceOf(localAccID)
-		if err != nil {
-			l.Error("failed to read GAS token balance of the local account, will try again later", zap.Error(err))
-			return
+		// deposit when balance falls below 1/5 of the single deposit amount
+		const refillProportion = 5
+
+		if new(big.Int).Mul(notaryBalance, big.NewInt(refillProportion)).Cmp(singleNotaryDepositAmount) >= 0 {
+			l.Info("enough funds on the notary balance, deposit is not needed", zap.Stringer("balance", notaryBalance))
```

Comment:
would not go `INFO` too

Reply:
will be resolved within #2419. For now it just helps to monitor current state

Reply:
yes, i just needed some reference to that line